### PR TITLE
Remove duplicated code

### PR DESCRIPTION
**Pull request by evilaliv3**
Hello @redshiftzero !

While retesting your ruleset generator i've found out the following:
- current jwk.py and pgppubkey_to_jwk.py implements same functionality with duplicated code and use files that have been renamed
- josepy can be used in place of authlib that seems to be adopted just to use their jose implementation. josepy is pretty more updated and minimal
- pgpy could be deprecated as josepy includes the same functionality currently needed by means of pgpy

### CODE REVIEW

**Comment by redshiftzero** [comment]
oh nice, TIL about josepy

File: pgppubkey_to_jwk.py [hunk 1]
```diff
@@ -1,17 +1,9 @@
-from authlib.common.encoding import int_to_base64
 import json
-import pgpy
+from josepy import JWK
 
+with open('public_release.pem', 'r') as f:
+    key = bytes(f.read(), 'utf-8')
 
-# Input is PGP armored pubkey
-key, _ = pgpy.PGPKey.from_file('key.asc')
```
Comment (redshiftzero) [inline pgppubkey_to_jwk.py:7]: so `jwk.py` is for taking a PEM encoded key and dumping to JWK: basically this is used by the target `make test-key` in this repo which makes the corresponding JWK format for experiment/testing purposes only

the idea with `pgppubkey_to_jwk.py` however was to take an existing PGP RSA public key (i.e. one not PEM formatted) and dump it in the expected JWK format (the idea there was that I wanted to use our existing SecureDrop PGP release key for the ruleset channel, and the extension required the JWK)

**Comment by evilaliv3** [comment]
Thank you for your feedback @redshiftzero 

I've revised the code that now should work also in other formats (i've tried in DER/binary)

**Comment by lepture** [comment]
Clarifies:

> josepy is pretty more updated

This is not true. Authlib is more updated with more algorithms.

> josepy is minimal

This is true, but Authlib is not that huge as you think. When you import `authlib.jose`, you will only use code in `jose` part.

But because `authlib.jose` contains **JWS**, **JWE**, **JWK**, **JWA**, and **JWT**, while josepy only contains **JWS**, **JWK**, and **JWA**. ``authlib.jose`` has more code than josepy, in this case, josepy is minimal than Authlib.

**Comment by evilaliv3** [comment]
Thank you so much @lepture for clarifying and my excuses, i was definitely wrong in that part of my comment!

I had this biased preference for josepy just because in this context it is currently already debian packaged and there my bias.

File: pgppubkey_to_jwk.py [hunk 2]
```diff
@@ -1,17 +1,9 @@
-from authlib.common.encoding import int_to_base64
 import json
-import pgpy
+from josepy import JWK
 
+with open('public_release.pem', 'rb') as f:
```
Comment (redshiftzero) [inline pgppubkey_to_jwk.py:4]: using an armored PGP public key as input here doesn't seem to work (output of the script ends up being a JSON object with  a `"kty": "RSA"` entry, but without `e`, `n` fields), is there something we need to do in josepy to handle the armored PGP public key case? 

if no that's fine, we can just keep this script as separate (its purpose is to convert pgp pubkey -> jwk object) and add back those required dependencies. 

basically we want to continue to handle two cases:
* one that takes PGP key as input, produces valid JWK (this script)
* one that takes PEM, produces valid JWK, and is used in `make test-key` (in `Makefile`)

**Comment by eloquence** [comment]
Hi @evilaliv3, thanks for your work on this. Just a friendly ping - do you have time to address the outstanding review comment in coming days?

**Comment by evilaliv3** [comment]
Actually no @eloquence, i'm so sorry, but actually after the feedback of @lepture i consider that we may discard my patch entirely.

**Comment by eloquence** [comment]
No worries @evilaliv3, closing this PR accordingly.
